README.md

```diff
@@ -1,19 +1,158 @@
+[![Coverage Status](https://coveralls.io/repos/github/Mellanox/nvidia-k8s-ipam/badge.svg)](https://coveralls.io/github/Mellanox/nvidia-k8s-ipam)
+[![Build, Test, Lint](https://github.com/Mellanox/nvidia-k8s-ipam/actions/workflows/build-test-lint.yml/badge.svg?event=push)](https://github.com/Mellanox/nvidia-k8s-ipam/actions/workflows/build-test-lint.yml)
+[![CodeQL](https://github.com/Mellanox/nvidia-k8s-ipam/actions/workflows/codeql.yml/badge.svg)](https://github.com/Mellanox/nvidia-k8s-ipam/actions/workflows/codeql.yml)
+[![Image push](https://github.com/Mellanox/nvidia-k8s-ipam/actions/workflows/image-push-main.yml/badge.svg?event=push)](https://github.com/Mellanox/nvidia-k8s-ipam/actions/workflows/image-push-main.yml)
 
-### Deploy IPAM
+An IP Address Management (IPAM) CNI plugin designed to operate in a Kubernetes environment.
+This Plugins allows to assign IP addresses dynamically across the cluster while keeping speed
+and performance in mind.
 
-_NOTE: This command will deploy latest dev build_
+NVIDIA IPAM plugin currently support only IP allocation for K8s Secondary Networks.
+e.g Additional networks provided by [multus CNI plugin](https://github.com/k8snetworkplumbingwg/multus-cni).
 
+This repository is in its first steps of development, APIs may change in a non backward compatible manner.
+
+## Introduction
+
+NVIDIA IPAM plugin consists of 3 main components:
+
+1. controller ([ipam-controller](#ipam-controller))
+2. node agent ([ipam-node](#ipam-node))
+3. IPAM CNI plugin ([nv-ipam](#nv-ipam))
+
+### ipam-controller
+
+A Kubernetes(K8s) controller that Watches on a predefined K8s ConfigMap for defined IP pools.
+It then proceeds in reconciling K8s Node objects by assiging each node via `ipam.nvidia.com/ip-blocks`
+annotation a cluster unique range of IPs of the defined pools.
+
+### ipam-node
+
+A node agent that performs initial setup and installation of nv-ipam CNI plugin.
+
+### nv-ipam
+
+An IPAM CNI plugin that allocates IPs for a given interface out of the defined IP pool as provided
+via CNI configuration.
+
+IPs are allocated out of the provided IP Block assigned by ipam-controller for the node.
+To determine the cluster unique IP block for the defined IP pool, nv-ipam CNI queries K8s API
+for the Node object and extracts IP block infromation from node annotation.
+
+nv-ipam plugin currently leverages [host-local](https://www.cni.dev/plugins/current/ipam/host-local/)
+IPAM to allocate IPs from the given range.
+
+### IP allocation flow
+
+1. User (cluster administrator) defines a set of named IP Pools to be used for IP allocation
+of container interfaces via Kubernetes ConfigMap (more information in [Configuration](#configuration) section)
+
+_Example_:
+
+```json
+{
+    "pools":  {
+        "my-pool": {"subnet": "192.188.0.0/16", "perNodeBlockSize": 24, "gateway": “192.168.0.1”},
```

Comment:
Invalid quotes the for gateway option.

Reply:
Also remove last , so it will be valid json

Reply:
ack, this was copied from our doc, i have fixed there as well.